bigtable/google/cloud/bigtable/row_set.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""User-friendly container for Google Cloud Bigtable RowSet """
+
+
+from google.cloud._helpers import _to_bytes
+
+
+class RowSet(object):
+    """ Convenience wrapper of google.bigtable.v2.RowSet
+        Useful for creating a set of row keys and row ranges, which can
+        be passed to yield_rows method of class:`.Table.yield_rows`.
+    """
+
+    def __init__(self):
+        self.row_keys = []
+        self.row_ranges = []
+
+    def add_row_key(self, row_key):
+        """Add row key to row_keys list.
+
+        :type row_key: bytes
+        :param row_key: The key of a row to read
+        """
+        self.row_keys.append(row_key)
+
+    def add_row_range(self, row_range):
+        """Add row_range to row_ranges list.
+
+        :type row_range: class:`RowRange`
+        :param row_range: The row range object having start and end key
+        """
+        self.row_ranges.append(row_range)
+
+    def _update_message_request(self, message):
+        """Add row keys and row range to given request message
+
+        :type message: class:`data_messages_v2_pb2.ReadRowsRequest`
+        :param message: The ``ReadRowsRequest`` protobuf
+        """
+        for each in self.row_keys:
+            message.rows.row_keys.append(_to_bytes(each))
+
+        for each in self.row_ranges:
+            r_kwrags = each.get_range_kwargs()
+            message.rows.row_ranges.add(**r_kwrags)
+
+
+class RowRange(object):
+    """ Convenience wrapper of google.bigtable.v2.RowRange
+
+    :type start_key: bytes
+    :param start_key: (Optional) Start key of the row range. If left empty,
+                      will be interpreted as the empty string.
+
+    :type end_key: bytes
+    :param end_key: (Optional) End key of the row range. If left empty,
+                    will be interpreted as the infinite string.
```

Comment:
I don't understand "infinite string" here.  I believe the semantics are that if not passed, then the range will be unbounded on the high end.

Reply:
Updated the doc string.